pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -330,6 +330,12 @@ stdenv.mkDerivation ({
         export NIX_${ghcCommandCaps}="${ghcEnv}/bin/${ghcCommand}"
         export NIX_${ghcCommandCaps}PKG="${ghcEnv}/bin/${ghcCommand}-pkg"
         export NIX_${ghcCommandCaps}_DOCDIR="${ghcEnv}/share/doc/ghc/html"
+        for f in $NIX_LDFLAGS; do
```

Comment:
`NIX_LDFLAGS` is not the right place to take this information from. That path list ought to be complied from `systemBuildInputs` instead.